common/app/com/commercetools/sunrise/ctp/graphql/SingleGraphQLRequest.java

```diff
@@ -0,0 +1,53 @@
+package com.commercetools.sunrise.ctp.graphql;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.sphere.sdk.client.HttpRequestIntent;
+import io.sphere.sdk.http.HttpMethod;
+import io.sphere.sdk.http.HttpResponse;
+import io.sphere.sdk.json.SphereJsonUtils;
+import io.sphere.sdk.models.Base;
+
+import javax.annotation.Nullable;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.joining;
+
+public abstract class SingleGraphQLRequest<T> extends Base implements GraphQLRequest<T> {
+
+    private final String fieldName;
+    private final TypeReference<T> typeReference;
+
+    protected SingleGraphQLRequest(final TypeReference<T> typeReference, final String fieldName) {
+        this.fieldName = fieldName;
+        this.typeReference = typeReference;
+    }
+
+    @Nullable
+    @Override
+    public T deserialize(final HttpResponse httpResponse) {
+        final JsonNode rootJsonNode = SphereJsonUtils.parse(httpResponse.getResponseBody());
+        if (rootJsonNode.has("data")) {
+            final JsonNode data = rootJsonNode.get("data");
+            if (data.has(fieldName)) {
+                return SphereJsonUtils.readObject(data.get(fieldName), typeReference);
+            } else {
+                throw new GraphQLException("No field " + fieldName + " in response", httpResponse);
+            }
+        } else {
+            throw new GraphQLException("No data found in response", httpResponse);
+        }
+    }
+
+    @Override
+    public HttpRequestIntent httpRequestIntent() {
+        final JsonNode variables = getVariables();
+        final String variablesPart = variables == null ? null : String.format("\"variables\": %s", variables);
```

Comment:
According to the javadoc [JsonNode#tostring()](http://fasterxml.github.io/jackson-databind/javadoc/2.8/com/fasterxml/jackson/databind/JsonNode.html#toString()) we shouldn't rely on `JsonNode#toString` method here. Instead we should use one of the helper methods of `SphereJsonUtils` here.